frontend/src/scenes/pipeline/ReplayMenu.tsx

```diff
@@ -0,0 +1,465 @@
+    retries: any[]
+    showLabel?: boolean
+}): JSX.Element {
+    const colorForStatus = (status: string): 'success' | 'primary' | 'warning' | 'danger' | 'default' => {
+        switch (status) {
+            case 'success':
+                return 'success'
+            case 'error':
+                return 'danger'
+            default:
+                return 'default'
+        }
+    }
+
+    const status = retries.some((retry) => retry.status === 'success') ? 'success' : 'error'
+    const color = colorForStatus(status)
+
+    return (
+        <Tooltip
+            title={
+                <>
+                    Run status: {status}
+                    {retries.length > 1 && (
+                        <>
+                            <br />
+                            Attempts: {retries.length}
+                        </>
+                    )}
+                </>
+            }
+        >
+            <span
+                className={clsx(
+                    `RetryStatusIcon h-6 p-2 border-2 flex items-center justify-center rounded-full font-semibold text-xs select-none`,
+                    color === 'primary' && 'RetryStatusIcon--pulse',
+                    showLabel ? '' : 'w-6',
+                    retries.length > 0 ? `border-${color} text-${color}-dark` : ''
+                )}
+            >
+                {showLabel ? <span className="text-center">{status}</span> : retries.length}
+            </span>
+        </Tooltip>
+    )
+}
+
+function EmptyColumn(): JSX.Element {
+    return (
+        <Tooltip title="NULL" placement="right" delayMs={0}>
+            <span className="cursor-default" aria-hidden>
+                —
+            </span>
+        </Tooltip>
+    )
+}
+
+function RunsFilters({ id }: { id: string }): JSX.Element {
+    const logic = hogFunctionReplayLogic({ id })
+    const { eventsLoading, baseEventsQuery } = useValues(logic)
+    const { loadEvents, changeDateRange, loadTotalEvents } = useActions(logic)
+
+    const handleRefresh = (): void => {
+        loadEvents()
+        loadTotalEvents()
+    }
+
+    return (
+        <div className="flex items-center gap-2">
+            <LemonButton
+                onClick={handleRefresh}
+                loading={eventsLoading}
+                type="secondary"
+                icon={<IconRefresh />}
+                size="small"
+            >
+                Refresh
+            </LemonButton>
+            <DateFilter
+                dateFrom={baseEventsQuery?.after ?? undefined}
+                dateTo={baseEventsQuery?.before ?? undefined}
+                onChange={changeDateRange}
+            />
+        </div>
+    )
+}
+
+export function HogFunctionEventEstimates({ id }: { id: string }): JSX.Element | null {
+    const logic = hogFunctionReplayLogic({ id })
+    const { eventsLoading, eventsWithRetries, loadingRetries, totalEvents } = useValues(logic)
+    const { retryHogFunction } = useActions(logic)
+
+    return (
+        <LemonTable
```

Comment:
I think we will need a description above making it clear that this is a list of the history of events and can be retried but that it is not an indication of whether it was previously successfully sent or not. Otherwise it will be super confusing as it appears like a history of runs when it really isn't.